lib/data/extract.rb

```diff
@@ -11,10 +11,10 @@
 districts = []
 townships = []
 
-CSV.foreach('locations.csv') do |iso, s_pcode, alpha3, state, d_pcode, district, t_pcode, township| # rubocop:disable Metrics/ParameterLists
-  states << { iso: iso, pcode: s_pcode, alpha3: alpha3, name: state }
-  districts << { pcode: d_pcode, name: district, state: s_pcode } unless d_pcode.nil?
-  townships << { pcode: t_pcode, name: township, district: d_pcode } unless t_pcode.nil?
+CSV.foreach('locations.csv') do |iso, s_pcode, alpha3, state, state_in_mm, d_pcode, district, district_in_mm, t_pcode, township, township_in_mm| # rubocop:disable Metrics/ParameterLists, Layout/LineLength
```

Comment:
- There must be a way we can capture all arguments with a splat like `*values`
- Let's use the next line to construct keys `keys = [:iso, :s_pcode, ...]`
- Then let's call `nayyar = Hash[keys, values]` to get a big hash of the line
- we can then populate `states`, `districts` and `townships` using the hash. 
- Let's also name the new attribute `my_name` since `my` is the language code for Myanmar

Reply:
I tried to construct the nayyar hash and populate using the hash. I find it a bit hard to read and have to break the line. Am I doing this correctly?
```ruby
  unless nayyar[:d_pcode].nil?
    districts << { pcode: nayyar[:d_pcode], name: nayyar[:district], my_name: nayyar[:district_in_my],
                   state: nayyar[:s_pcode] }
  end
```

Is splatting and assigning it to variables a good way to do it? It will make the code shorter.
```ruby
  iso, s_pcode, alpha3, state, state_in_my, d_pcode, district, district_in_my, t_pcode, township,
              township_in_my = *values
```

Reply:
How about you do it like this?

```
CSV.foreach('locations.csv') do |*location_csv_data|
  keys = [:pcode, :name, :etc]
  location = Hash[key, location_csv_data]
  states << location if state?(location)
  districts << location if district?(location)
  townships << location if township?(location)
end
```

Reply:
Yes. This is more neat!